Adder timer function and option for filename

Dark-Kernel opened the pull request:
1. Developed a small timer function, which will run after running the script to check the length during recording.
2. Added -o option which takes the filename as an argument, to save the file with the name you want
3. Included DIR entry in configuration to change the path, where the file will be saved.

Dark-Kernel committed:
Developed -o option for output filename & included DIR for file path in configuration.

Dark-Kernel committed:
Added timer to recording.

belden commented on bin/ffscreencast:
```diff
@@ -40,7 +40,7 @@ TIME=$(date +%H.%M.%S)
 
 # Where to save it
-DIR="${HOME}/Desktop"
+DIR="${HOME}/Videos/"
```
@Dark-Kernel Is this a mistaken commit? I imagine it might be left over from your development, since it's not called out in the PR description.

Dark-Kernel replied:
Sorry but, No it's not mistaken. Is there something wrong?

belden replied:
To be very clear: I think this is a good change to make, and might resolve https://github.com/cytopia/ffscreencast/issues/34 (or might not, perhaps that user doesn't keep a `~/Videos/` either).

This is a change in prior behavior from how the tool used to operate. It might be worth making a separate PR just for changing the default save directory, so @cytopia can either approve or reject it as they see fit, independent of the timer functionality your PR is adding.

Dark-Kernel replied:
Videos directory is most likely available on all Desktop Linux distributions, and if it's not there that means most probably that user will neither have Desktop directory. (Like me)
BTW thanks.

belden commented on bin/ffscreencast:
```diff
@@ -153,52 +179,58 @@ write_config() {
+			echo 
+
+			echo "# Default Directory to save video files" 
+			echo "DIR=\"$HOME/Videos\"" 
```
Again I wonder whether this is an intended change?

Dark-Kernel replied:
Having DIR in configuration, one can easily change it to intended location.

belden replied:
Just to reply here as well, changing default behavior of tools can surprise users. Though I personally think `~/Videos/` is a more sensible default location than `~/Desktop/`, I have also gotten used to the existing behavior of the tool.

Dark-Kernel replied:
I agree with you, so I think I should change it to ~/Desktop again.

Sorry for the inconvenience.

belden commented on bin/ffscreencast:
```diff
@@ -153,52 +179,58 @@ write_config() {
+			echo "CHOSEN_A_NUM=\"\""
+			echo "CHOSEN_C_NUM=\"\""
+			echo 
+		} >> "${conf}"
```
@Dark-Kernel FYI there's a subtle change in behavior here. Previously passing through this code would do

    echo "foo" > $conf # clobber

and then a series of

    echo "bar" >> $conf # append new file

Changing this to

    {
        echo "foo"
        echo "bar"
    } >> $conf

means we no longer clobber the old config, instead we'll append a new config to it.

I think you can just use

    {
        echo "foo"
        echo "bar"
    } > $conf # note, just a single `>`

to achieve the prior behavior.

Dark-Kernel replied:
Yes, it's small mistake, i didn't noticed that one >  is extra, sorry 😅.

belden replied:
@Dark-Kernel cool, it seems like an easy fix - but this PR doesn't contain it (yet - maybe you haven't had a chance to make it yet: totally understandable!)

belden replied:
I've opened https://github.com/Dark-Kernel/ffscreencast/pull/1 which targets your branch - if merged it will resolve this `>>` -> `>` issue.

belden commented on bin/ffscreencast:
```diff
@@ -1053,7 +1093,7 @@ if [ "${RECORD_S}" = "yes" ]; then
 		if ! screen_device_exists "$CHOSEN_S_NUM"; then
 			echo "Screen recording device: '${CHOSEN_S_NUM}' does not exist."
-			exit -1
+			exit 1
```
@Dark-Kernel I'm curious about this change? Again, it's a slight change, but it's not commented in the PR so I'm uncertain as to the rationale.

```sh
$ bash -c 'exit -1'
$ echo $?
255

$ bash -c 'exit 1'
$ echo $?
1
```

Dark-Kernel replied:
what difference it makes?

belden replied:
Again, this is just me noticing an unacknowledged change in default behavior.

My interest in this project is to be able to easily record my screen on Linux. I use `i3wm`, which is a very customizable window manager.

Right now it is nice that `ffscreencast` has three different exit values:

* `0` -- no errors; my screen recording is where I want it to be
* `1` -- some sort of error occurred, probably while running
* `255` -- the script couldn't even start doing its work

Changing from `exit -1` to `exit 1` here means I get a little less information from the exit value of `ffscreencast`.

Dark-Kernel replied:
Cool! Learned something new. Will change it.

belden commented on bin/ffscreencast:
```diff
@@ -859,6 +894,11 @@ while [ $# -gt 0  ]; do
+		-o*)
+			shift
+			NAME="$1" 
+            run_timer &
```
Nit, this line is indented with spaces rather than tabs. The file in general uses tabs.

Dark-Kernel replied:
This I saw in git diff, I always use tabs. I don't know how this occurred.

belden replied:
I've opened https://github.com/Dark-Kernel/ffscreencast/pull/1 which targets your branch - if merged, it will resolve this issue.

Dark-Kernel replied:
Done.

belden reviewed (commented):
Some questions and minor concerns about possible unintended changes.

belden committed:
Ensure writing config clobbers old config

This preserves the prior behavior of the script. See
https://github.com/cytopia/ffscreencast/pull/39#discussion_r1678104297

belden committed:
Whitespace cleanup

No logic change

* delete trailing spaces
* swap a space-indent to a hard-tab indent

belden commented on bin/ffscreencast:
```diff
@@ -273,6 +306,8 @@ print_help() {
 	echo "                  Use: --oargs=\"-crf 0\""
 	echo "                  Default: '-crf 0 -preset ultrafast'"
+	echo 
+	echo "-o           Output video file name"
```
I think this option should document its usage, similar to how the others do.

Dark-Kernel replied:
Yeah, Today itself this pull request will be merged/rejected.

Dark-Kernel committed:
Merge pull request #1 from belden/pr-tidy

Tidy whitespace and clobber config

Dark-Kernel committed:
changed exit status back to -1, Default dir o desktop.  Added usage for -o option

Dark-Kernel commented:
@belden 
I request you to complete this please.

belden commented:
@Dark-Kernel I wish that I could! I am not a contributor on this project - just another programmer out in the wild who also has some open PRs for this project.

I've tried to help this PR along to make sure there aren't any broken backwards compatibility issues, but it's up to @cytopia to ultimately review and approve/reject this work.

@cytopia can you have a look at this PR please?
